packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4329,6 +4332,42 @@ class InheritedElement extends ProxyElement {
     setDependencies(dependent, null);
   }
 
+  /// Called by [dependent] to stop listening to this [InheritedElement].
+  ///
+  /// Subclasses can override this method to cancel potential side-effects
+  /// related to subscriptions.
```

Comment:
This would be a good place to mention hasDependencies.  I assume the use case we're enabling is checking hasDependencies after calling super.removeDependencies().

Reply:
Indeed, done. Is that what you had in mind?